File: UI/wwwroot/Facturacion/FrontModel/Detalle_Factura.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Producto } from './Cat_Producto.js';
import { Tbl_Factura } from './Tbl_Factura.js';
import { Tbl_Lotes } from "./Tbl_Lotes.js";
class Detalle_Factura extends EntityClass {
    /**
    * @param {Partial<Detalle_Factura>} props 
    */
    constructor(props) {
        super(props, 'EntityFacturacion');
        Object.assign(this, props);
    }
    /**@type {Number}*/ Id_DetalleFactura;
    /**@type {Number}*/ Cantidad;
    /**@type {Number}*/ Precio_Venta;
    /**@type {Number}*/ Iva;
    /**@type {Number}*/ Total;
    /**@type {Number}*/ Id_Lote;
    /**@type {Number}*/ Descuento;
    /**@type {Number}*/ Monto_Descuento;
    /**@type {Number}*/ Sub_Total;
    /**@type {Tbl_Lotes} ManyToOne*/ Lote;
    //**@type {Tbl_Factura} ManyToOne*/ Tbl_Factura;
    //**@type {Cat_Producto} ManyToOne*/ Cat_Producto;
}
export { Detalle_Factura };

File: BusinessLogic/Facturacion/Mapping/Detalle_Factura.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using APPCORE;
namespace DataBaseModel {
	public class Detalle_Factura : EntityClass
		{
				[PrimaryKey(Identity = true)]
				public int? Id_DetalleFactura { get; set; }
				public int? Id_Factura { get; set; }
				public int? Id_Producto { get; set; }
				public double? Cantidad { get; set; }
				public double? Precio_Venta { get; set; }
				public double? Iva { get; set; }
				public double? Total { get; set; }
				public int? Id_Lote { get; set; }
				public double? Descuento { get; set; }
				public double? Sub_Total { get; set; }				
				public double? Monto_Descuento { get;  set; }
				
				//[ManyToOne(TableName = "Tbl_Factura", KeyColumn = "Id_Factura", ForeignKeyColumn = "Id_Factura")]
				public Tbl_Factura? Tbl_Factura { get; set; }
				
				[ManyToOne(TableName = "Tbl_Lotes", KeyColumn = "Id_Lote", ForeignKeyColumn = "Id_Lote")]
				public Tbl_Lotes? Lote { get; set; }
		}
}
